src/data/items.h

```diff
@@ -3513,560 +3513,560 @@ const struct Item gItems[] =
 
 // TMs/HMs
 
-    [ITEM_TM01_FOCUS_PUNCH] =
+    [ITEM_TM01] =
```

Comment:
The price and description are still linked to the specific TM/HM, so if you're aiming for fewest things to update these should probably stay move-specific. That would just leave the item name.

Reply:
I'm still trying to work out a good approach to doing this. Ideally I think in this file the constants would still be `ITEM_TM01_FOCUS_PUNCH` etc etc, but I don't have a nice way to make `FOREACH_TM`/`FOREACH_HM` tag each element with a number that can be `##`ed onto the names. Perhaps I might be able to do something with `__COUNTER__`, although then I'll still want to look for a way to pad 1-9 with a leading `0`.

Reply:
Ok, well I think this is fine otherwise. I'm not worried about it being fully automated like that, so if it's a choice between e.g. `ITEM_TM01` and `ITEM_TM_FOCUS_PUNCH` here I'd go with the latter.

Reply:
Pushed this change! :)